### DISCUSSION TITLE
Python UDFs

### DISCUSSION THREAD

**Discussion by scottee** [duckdb/duckdb#3202] [Q&A]
I see several references in discussions and release notes to Python UDFs. Is there an example of what it can do and how it would be used?

Ideally, my use case is to define python scalar or aggregate functions and use it from sql. I can do this in the sqllite3 python wrapper easily with its "create_function()" and "create_aggregate()" calls, so I was hoping it might also be available in duckdb.

**Comment by hannes** [comment]
There is indeed little documentations on this yet, but have a look here: 
https://github.com/duckdb/duckdb/blob/master/tools/pythonpkg/tests/fast/test_map.py

**Reply by mauropagano** [reply]
@hannes trying to use this I see my function called every 1024 (`STANDARD_VECTOR_SIZE`) rows, that's the expected behavior right? Is there a way to increase the number (ideally call the function once per relation) ?

**Reply by hannes** [reply]
No, that is the expected behaviour. There is intentionally no way of calling the function on the entire dataset, we made this mistake in the past.

**Reply by mauropagano** [reply]
Thanks! Makes sense, I can see that being a terrible idea more often than not
